### ISSUE TITLE
Empty route with headers not possible?

### ISSUE THREAD

**Issue by edsko** [haskell-servant/servant#239]
A route of type 

``` haskell
Post '[] ()
```

works fine, but a route of type

``` haskell
Post '[] (Headers '[Header "X-An-Int" Int] ())
```

yields

```
    Could not deduce (Servant.API.ContentTypes.IsNonEmpty '[])
      arising from a use of ‘serve’
```

**Comment by jkarni** [comment]
Yeah, I guess we're missing a few overlapping instances for `Headers ls ()`. Should be a simple fix, but I really want to focus on the HTTP errors work first, so I might not have time to fix it this weekend.

**Comment by jkarni** [comment]
(And thanks for the report!)

**Comment by edsko** [comment]
Sure, I'm not in any way blocked on this.

**Comment by jkarni** [comment]
It's easy enough to fix this by adding new instances, but we'd be going further down the route of exponential blow-up in instances. Not sure if this is the right ticket for it, but definitely something this brings up...

**Comment by fizruk** [comment]
@jkarni I think this is also closed by #276 and `NoContent` solution.
